src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -21,120 +103,808 @@ const ActorInit En_Pamera_InitVars = {
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06008448, &D_060005BC, this->limbDrawTbl, this->transitionDrawTbl,
+                     0x17);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, NULL, &sColChkInfoInit2);
+    this->actor.targetMode = 6;
+    this->unk_312 = 0;
+    this->unk_310 = 0;
+    this->unk_314 = 0;
+    this->unk_316 = 0;
+    this->unk_31C = 0;
+    this->unk_322 = 0;
+    this->unk_324 = 0;
+    if (((this->actor.params & 0xF000) >> 0xC)) {
+        func_80BD9840(this, globalCtx);
+    } else {
+        func_80BD8588(this, globalCtx);
+        func_80BD8658(this);
+        if (1) {}
+        if (((gSaveContext.weekEventReg[14] & 4) == 0) || ((gSaveContext.weekEventReg[52] & 0x20) != 0) ||
+            ((gSaveContext.weekEventReg[75] & 0x20) != 0) || (gSaveContext.entranceIndex == 0x2090)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+        if ((gSaveContext.weekEventReg[61] & 4) != 0) {
+            if (((gSaveContext.weekEventReg[59] & 1) == 0) || (gSaveContext.entranceIndex != 0x2020)) {
+                if ((gSaveContext.entranceIndex != 0x2020) && ((gSaveContext.weekEventReg[59] & 1) != 0)) {
+                    gSaveContext.weekEventReg[59] &= 0xFE;
+                }
+                func_80BD8700(this);
+            } else {
+                this->pathIndex = this->pathPointsCount - 1;
+                Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->pathPoints[this->pathIndex]);
+                Math_Vec3s_ToVec3f(&sp44, &this->pathPoints[this->pathIndex - 1]);
+                this->actor.world.rot.y = this->actor.shape.rot.y = Math_Vec3f_Yaw(&sp44, &this->actor.world.pos);
+                func_80BD8CCC(this);
+            }
+        } else {
+            gSaveContext.weekEventReg[59] |= 1;
+            func_80BD8FF0(this);
+        }
+    }
+}
+
+s32 func_80BD84F0(EnPamera* this, GlobalContext* globalCtx) {
+    Actor* actorPtr = globalCtx->actorCtx.actorList[ACTORCAT_DOOR].first;
+
+    while (actorPtr != 0) {
+        if ((actorPtr->id == ACTOR_EN_DOOR) && (Math_Vec3f_DistXZ(&this->actor.world.pos, &actorPtr->world.pos) < 200.0f)) {
+            this->actor.child = actorPtr;
+            return 1;
+        }
+        actorPtr = actorPtr->next;
+    }
+    this->actor.child = NULL;
+    return 0;
+}
+
+void func_80BD8588(EnPamera* this, GlobalContext* globalCtx) {
+    Path* path = &globalCtx->setupPathList[((this->actor.params & 0xFF0) >> 4)];
+    Vec3f sp28;
+
+    if (path == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    if (gSaveContext.weekEventReg[61] & 4) {
+        path = &globalCtx->setupPathList[(u8)path->unk1];
+    }
+    this->pathPoints = Lib_SegmentedToVirtual(path->points);
+    this->pathIndex = 0;
+    this->pathPointsCount = path->count;
+    this->animatedTextures = (u8)path->unk1;
```

Comment:
Looks like `unk1` in that struct has already been changed to a `u8`, so the cast shouldn't be necessary.